tests/ondemand/ondemand_basictests.cpp

```diff
@@ -181,47 +180,66 @@ namespace number_tests {
   // adapted by D. Lemire
   inline uint64_t f64_ulp_dist(double a, double b) {
     uint64_t ua, ub;
-    memcpy(&ua, &a, sizeof(ua));
-    memcpy(&ub, &b, sizeof(ub));
-    if ((int64_t)(ub ^ ua) >= 0)
-      return (int64_t)(ua - ub) >= 0 ? (ua - ub) : (ub - ua);
+    std::memcpy(&ua, &a, sizeof(ua));
+    std::memcpy(&ub, &b, sizeof(ub));
+    if (static_cast<int64_t>(ub ^ ua) >= 0)
+      return static_cast<int64_t>(ua - ub) >= 0 ? (ua - ub) : (ub - ua);
     return ua + ub + 0x80000000;
   }
 
   bool small_integers() {
     std::cout << __func__ << std::endl;
     for (int64_t m = 10; m < 20; m++) {
       for (int64_t i = -1024; i < 1024; i++) {
-        return test_ondemand<int64_t>(std::to_string(i), [&](int64_t actual) {
+        auto test=[&](int64_t actual) {
```

Comment:
How necessary is it to pull the lambda out of the call? Most of the other tests leave it as part of the test_ondemand call, so this will look a tiny bit out of place.

Reply:
I think it made it easier to read (nesting level) - but sure, could be rewritten. Do you think the new version makes it easier to spot the (erroneous) early return?

Reply:
I undid the lambda extraction.